core/app/models/spree/product.rb

```diff
@@ -161,6 +161,11 @@ def discontinued?
       !!discontinue_on && discontinue_on <= Time.current
     end
 
+    # determine if any variant (including master) can be supplied
+    def can_supply?
+      variants_including_master.any?(&:can_supply?)
```

Comment:
I think that if there are any variants, we shouldn't include master variant in this check. Master variant keeps default count_on_hand = 10 even if all other variants are sold.